### ISSUE TITLE
Saving markdown files via the Web interface fails.

### ISSUE THREAD

**Issue by sgtwilko** [go-gitea/gitea#2693]
- Gitea version (or commit ref): 1.2.0 built with: bindata, sqlite
- Git version: 2.14.1
- Operating system: Windows server 2008R2
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:

No errors listed in the log, but please let me know if there's some extra logging I can enabled that would provide information.

## Description

When trying to edit Markdown files (such as README.md) via the web interface we receive this error:
`Failed to update/create file 'README.md' with error: UpdateLocalCopyBranch [branch: Development]: git clone Development: exit status 128 - fatal: destination path 'data/tmp/local-repo/2' already exists and is not an empty directory.`

## Screenshots

![gitea_markdownsaveerror](https://user-images.githubusercontent.com/658876/31502291-87b7af14-af64-11e7-9e5f-c01124956eec.PNG)

**Comment by lafriks** [comment]
I think this could be windows related...

**Comment by sgtwilko** [comment]
After a fair bit of time with Process Monitor (thank you Sysinternals!) I've found the issue.

For some reason (don't know why, I didn't install either) we had a 32bit version of Git and the 64bit version of Gitea.

Gitea was looking in System32 for the `data/tmp/local-repo` folder, Git was looking in what it thought was System32, but a 32bit application on a 64bit system is remapped by the OS to point to the SysWOW64, so the folder was never removed.

I've uninstalled the 32bit version and installed the 64bit and the issue went away.

Probably Gitea needs to either check that the bit versions match, but a better fix would be to force the data/tmp/ folder to be in the Gitea install location rather than in System32.  I can't think of a reason why Gitea needs to be using the System32 folder for temp storage.
